wrapper/src/test/java/integration/util/AuroraTestUtility.java

```diff
@@ -344,14 +430,342 @@ public void deleteCluster() {
+      case "aurora-mysql":
+        return DatabaseEngine.MYSQL;
+      default:
+        throw new UnsupportedOperationException(response.dbClusters().get(0).engine());
+    }
+  }
+
+  public List<TestInstanceInfo> getClusterInstanceIds(final String clusterId) {
+    final DescribeDbInstancesResponse dbInstancesResult =
+        rdsClient.describeDBInstances(
+            (builder) ->
+                builder.filters(Filter.builder().name("db-cluster-id").values(clusterId).build()));
+
+    List<TestInstanceInfo> result = new ArrayList<>();
+    for (DBInstance instance : dbInstancesResult.dbInstances()) {
+      result.add(
+          new TestInstanceInfo(
+              instance.dbInstanceIdentifier(),
+              instance.endpoint().address(),
+              instance.endpoint().port()));
+    }
+    return result;
+  }
+
+  public void waitUntilClusterHasRightState(String clusterId) throws InterruptedException {
+    String status = getDBCluster(clusterId).status();
+    while (!"available".equalsIgnoreCase(status)) {
+      TimeUnit.MILLISECONDS.sleep(1000);
+      status = getDBCluster(clusterId).status();
+    }
+  }
+
+  public DBCluster getDBCluster(String clusterId) {
+    final DescribeDbClustersResponse dbClustersResult =
+        rdsClient.describeDBClusters((builder) -> builder.dbClusterIdentifier(clusterId));
+    final List<DBCluster> dbClusterList = dbClustersResult.dbClusters();
+    return dbClusterList.get(0);
+  }
+
+  public List<String> getAuroraInstanceIds() throws SQLException {
+    return getAuroraInstanceIds(
+        TestEnvironment.getCurrent().getInfo().getRequest().getDatabaseEngine(),
+        ConnectionStringHelper.getUrl(),
+        TestEnvironment.getCurrent().getInfo().getDatabaseInfo().getUsername(),
+        TestEnvironment.getCurrent().getInfo().getDatabaseInfo().getPassword());
+  }
+
+  public List<String> getAuroraInstanceIds(
+      DatabaseEngine databaseEngine, String connectionUrl, String userName, String password)
+      throws SQLException {
+
+    String retrieveTopologySql;
+    switch (databaseEngine) {
+      case MYSQL:
+        retrieveTopologySql =
+            "SELECT SERVER_ID, SESSION_ID FROM information_schema.replica_host_status "
+                + "ORDER BY IF(SESSION_ID = 'MASTER_SESSION_ID', 0, 1)";
+        break;
+      case PG:
+        retrieveTopologySql =
+            "SELECT SERVER_ID, SESSION_ID FROM aurora_replica_status() "
+                + "ORDER BY CASE WHEN SESSION_ID = 'MASTER_SESSION_ID' THEN 0 ELSE 1 END";
+        break;
+      default:
+        throw new UnsupportedOperationException(databaseEngine.toString());
+    }
+
+    ArrayList<String> auroraInstances = new ArrayList<>();
+
+    try (final Connection conn = DriverManager.getConnection(connectionUrl, userName, password);
+        final Statement stmt = conn.createStatement()) {
+      // Get instances
+      try (final ResultSet resultSet = stmt.executeQuery(retrieveTopologySql)) {
```

Comment:
nit: should we just add the ResultSet declaration to the outer try block started at line 523?